### DISCUSSION TITLE
How to properly clear storage?

### DISCUSSION THREAD

**Discussion by ghost** [Cookie-AutoDelete/Cookie-AutoDelete#1100] [Q&A / Support] [answered]
I have about 50 websites in the whitelist, and I want to clear all the data of the rest.
How do I properly set everything up to achieve this?
I'm using Firefox.

I've enabled "Enable Automatic Cleaning", "Enable Greylist Cleanup on Browser Restart" and everything in "Other Browsing Data Cleanup Options".
And yet when I look at the "Manage Cookies and Sites Data" submenu in Firefox Settings, I see a lot of websites, that indeed have 0 cookies, but have sometimes up to 50 MB in storage.
And profile folder (storage/default) has many folders with just ".metadata-v2" in them.
Only if I'll add the website into "Exceptions - Cookies and Site Data" in Firefox Settings to "block", that website will finally not store anything on my computer.

**Comment by kennethtran93** [comment, marked as answer]
Unfortunately the webextension API, notably the browsingData section doesn't fully have the capability to fully remove everything, especially when third party sites do NOT set a cookie but utilizes some other form of storage.  We cannot iterate anything in any storage other than cookies, which is why we set our own cookie to every site you visit (first party only).  Tracking third-party sites, especially those that do not set cookies, require additional permissions (webRequest) to which we have yet to implement.

It is out of our hands at this point until somebody actually implements the function that will clean more items.  Also according to this bug report: https://bugzilla.mozilla.org/show_bug.cgi?id=1526246, it appears that a few things aren't cleaned up properly when clearing certain data types.

Hope this sheds some light into the issue.

Chrome does have the ability to clear more data types, but those have yet to be implemented at this time.
